pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -1409,10 +1409,17 @@ public boolean isValid(int timeout) throws SQLException {
     if (isClosed()) {
       return false;
     }
+    boolean changedNetworkTimeout = false;
     try {
-      int savedNetworkTimeOut = getNetworkTimeout();
+      int oldNetworkTimeout = getNetworkTimeout();
+      int newNetworkTimeout = timeout * 1000;
       try {
-        setNetworkTimeout(null, timeout * 1000);
+        // change network timeout only if the new value is more restrictive than the old one
+        // (zero means no timeout)
+        if (newNetworkTimeout != 0 && (oldNetworkTimeout == 0 || oldNetworkTimeout > newNetworkTimeout)) {
```

Comment:
Instead of using the word restrictive can we change the comment to "the new value is less than the current"
Also it's easier for me to read new < old

Reply:
"the new value is less than the current" is not exactly accurate because of the zero-timeout case, so I tried to come up with a more "creative" wording :) OK, I'll change that

Reply:
just put a note that 0 is really no timeout.

Reply:
changed